test/helpers/adapter_helper.rb

```diff
@@ -0,0 +1,27 @@
+module Semian
+  module AdapterTest
+    include Semian::Adapter
+
+    def semian_identifier
+      :semian_adapter_test
+    end
+
+    def raw_semian_options
+      @client_options
+    end
+  end
+
+  class AdapterTestClient
+    include AdapterTest
+
+    def initialize(**args)
+      @client_options = args.merge(success_threshold: 1,
+                                   error_threshold: 1,
+                                   error_timeout: 1)
+    end
+
+    def ==(other)
+      inspect == other.inspect
```

Comment:
@fw42 that's because when we do comparisons, it's between a strong and a weak reference to the same object.

comparing them via inspect to assert that it's the same object, even though the ID doesn't match.